test/request.test.js

```diff
@@ -170,3 +170,31 @@ test('should throw when setting request id twice', function (t) {
     });
 });
 
+test('should provide route info', function (t) {
```

Comment:
I think it might be more accurate to say route object here - I think inevitably we will want the router will expose some sort of first class route object.